ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,425 @@
+
+        tue::serialization::InputArchive a_in(f_rgbd);
+        rgbd::deserialize(a_in, *image);
+    }
+
+    // Read sensor pose
+    if (!ed::deserialize(r, "sensor_pose", sensor_pose))
+    {
+        std::cerr << "No field 'sensor_pose' specified." << std::endl;
+        return false;
+    }
+
+//    if (r.hasError())
+//    {
+//        std::cout << "Error while reading file '" << filename << "':\n\n" << r.error() << std::endl;
+//        return false;
+//    }
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool loadWorldModel(const std::string& model_name, ed::WorldModel& world_model)
+{
+    ed::UpdateRequest req;
+
+    ed::models::ModelLoader model_loader;
+
+    std::stringstream error;
+    if (!model_loader.create("_root", model_name, req, error, true))
+    {
+        std::cerr << "Model '" << model_name << "' could not be loaded:" << std::endl << std::endl;
+        std::cerr << error.str() << std::endl;
+        return false;
+    }
+
+    // Reset world
+    world_model = ed::WorldModel();
+
+    // Update world
+    world_model.update(req);
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+void usage()
+{
+    std::cout << "Usage: ed_segmenter IMAGE-FILE-OR-DIRECTORY  WORLDMODEL_NAME  ENTITY_ID" << std::endl;
+}
+
+// Getting roll, pitch and yaw from a quaternion,
+// copied from https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+
+struct Quaternion {
+    double w, x, y, z;
+};
+
+struct EulerAngles {
+    double roll, pitch, yaw;
+};
+
+EulerAngles ToEulerAngles(Quaternion q) {
+    EulerAngles angles;
+
+    // roll (x-axis rotation)
+    double sinr_cosp = 2 * (q.w * q.x + q.y * q.z);
+    double cosr_cosp = 1 - 2 * (q.x * q.x + q.y * q.y);
+    angles.roll = std::atan2(sinr_cosp, cosr_cosp);
+
+    // pitch (y-axis rotation)
+    double sinp = 2 * (q.w * q.y - q.z * q.x);
+    if (std::abs(sinp) >= 1)
+        angles.pitch = std::copysign(M_PI / 2, sinp); // use 90 degrees if out of range
+    else
+        angles.pitch = std::asin(sinp);
+
+    // yaw (z-axis rotation)
+    double siny_cosp = 2 * (q.w * q.z + q.x * q.y);
+    double cosy_cosp = 1 - 2 * (q.y * q.y + q.z * q.z);
+    angles.yaw = std::atan2(siny_cosp, cosy_cosp);
+
+    return angles;
+}
+
+double getYaw(const geo::Mat3& rotation)
+{
+    // Get quaternion
+    geo::Quaternion geo_quaternion;
+    rotation.getRotation(geo_quaternion);
+
+    // Convert it to struct
+    Quaternion quaternion;
+    quaternion.x = geo_quaternion.getX();
+    quaternion.y = geo_quaternion.getY();
+    quaternion.z = geo_quaternion.getZ();
+    quaternion.w = geo_quaternion.getW();
+
+    // Get the Euler angles
+    EulerAngles angles = ToEulerAngles(quaternion);
+    ROS_DEBUG_STREAM("Matrix: " << rotation << " --> yaw: " << angles.yaw);
+    return angles.yaw;
+}
```

Comment:
Quaternions can already be found somewhere in geolib. Other functionality, if missing, should be moved to geolib. As it is very likely somebody will need similar functionality in the future and might create very similar code somewhere in our code base.

Reply:
After making use of the projectTo2D function. these functions are no longer needed